src/test/java/org/folio/edge/core/ResponseCompressionTest.java

```diff
@@ -0,0 +1,143 @@
+                .put(SYS_OKAPI_URL, "http://localhost:" + okapiPort)
+                .put(SYS_SECURE_STORE_PROP_FILE, "src/main/resources/ephemeral.properties")
+                .put(SYS_LOG_LEVEL, "TRACE")
+                .put(SYS_REQUEST_TIMEOUT_MS, requestTimeoutMs)
+                .put(SYS_RESPONSE_COMPRESSION, true);
+
+        final DeploymentOptions opt = new DeploymentOptions().setConfig(jo);
+        vertx.deployVerticle(EdgeVerticle2Test.TestVerticle.class.getName(), opt, context.asyncAssertSuccess());
+
+        RestAssured.baseURI = "http://localhost:" + serverPort;
+        RestAssured.port = serverPort;
+        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+    }
+
+    @AfterClass
+    public static void tearDownOnce(TestContext context) {
+        logger.info("Shutting down server");
+        vertx.close(res -> {
+            if (res.failed()) {
+                logger.error("Failed to shut down edge-rtac server", res.cause());
+                fail(res.cause().getMessage());
+            } else {
+                logger.info("Successfully shut down edge-rtac server");
+            }
+
+            logger.info("Shutting down mock Okapi");
+            mockOkapi.close();
+        });
+    }
+
+
+    @Test
+    public void testResponseCompression(TestContext context) {
+        logger.info("=== Test response compression (Accept-Encoding: gzip / Accept-Encoding: deflate)  ===");
+
+        for (DecoderConfig.ContentDecoder type : DecoderConfig.ContentDecoder.values()) {
+            final Response resp = RestAssured.given()
+                    .config(RestAssured.config().decoderConfig(decoderConfig().contentDecoders(type)))
+                    .get("/admin/health")
+                    .then()
+                    .contentType(TEXT_PLAIN)
+                    .statusCode(200)
+                    .header(HttpHeaders.CONTENT_ENCODING.toString(), type.name().toLowerCase())
+                    .extract()
+                    .response();
+            assertEquals("\"OK\"", resp.body().asString());
+        }
+    }
+
+    @Test
+    public void testResponseNoCompressionHeaderInstance(TestContext context) {
+        logger.info("=== Test no compression (Accept-Encoding: instance) ===");
+
+        final Response resp = RestAssured.given()
+                .config(RestAssured.config().decoderConfig(decoderConfig().noContentDecoders()))
+                .header(new Header(org.apache.http.HttpHeaders.ACCEPT_ENCODING, "instance"))
+                .get("/admin/health")
+                .then()
+                .contentType(TEXT_PLAIN)
+                .statusCode(200)
+                .extract()
+                .response();
+        assertEquals("\"OK\"", resp.body().asString());
+    }
+
+    @Test
+    public void testResponseNoCompressionWithoutAcceptEncoding(TestContext context) {
+        logger.info("=== Test no compression (without Accept-Encoding header) ===");
+
```

Comment:
This test, apart from the name and log message, is the same as the one above (`testResponseNoCompressionHeaderInstance`). Based on the name, it appears this test not have the `Accept-Encoding` header below.

Reply:
@mreno-EBSCO, sure - you are totally correct. This is already fixed. Thanks.